opcua_tools/memory_optimizer.py

```diff
@@ -0,0 +1,40 @@
+"""
+Call the memory optimizer in order to minimize possible pandas' memory leaks.
+Works for Linux. Does not change anything for MacOS.
+"""
+import logging
+import sys
+from ctypes import cdll, CDLL
+
+import pandas as pd
+
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+
+__old_pandas_del = getattr(pd.DataFrame, "__del__", None)
+
+try:
+    cdll.LoadLibrary("libc.so.6")
+    libc = CDLL("libc.so.6")
+    libc.malloc_trim(0)
+except (OSError, AttributeError):
+    libc = None
+
+
+def __malloc_trim_del(self):
+    if __old_pandas_del:
+        __old_pandas_del(self)
+    libc.malloc_trim(0)
+
+
+def replace_default_pandas_del_method():
+    if libc:
+        logger.info("Applying memory optimizer for pd.DataFrame.__del__")
+        pd.DataFrame.__del__ = __malloc_trim_del
+    else:
+        print(
```

Comment:
Could you switch this out with "logger"? Trying to get rid of all possible prints

Reply:
Good catch, I missed that one!